code/datums/components/vision_hurting.dm

```diff
@@ -0,0 +1,27 @@
+/// A component that damages eyes that look at the owner
+/datum/component/vision_hurting
+	var/damage_per_second
+	var/message
+
+/datum/component/vision_hurting/Initialize(damage_per_second=1, message="Your eyes burn as you look at")
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.damage_per_second = damage_per_second
+	src.message = message
+
+	START_PROCESSING(SSdcs, src)
+
+/datum/component/vision_hurting/process(seconds_per_tick)
+	var/damage = damage_per_second * seconds_per_tick
+	for(var/mob/living/carbon/viewer in viewers(parent))
+		if(viewer.is_blind() || viewer.get_eye_protection() >= damage_per_second)
+			continue
+		var/obj/item/organ/internal/eyes/burning_orbs = locate() in viewer.organs
+		if(!burning_orbs)
+			continue
+		burning_orbs.apply_organ_damage(damage)
+		if(prob(max(5, 100 - (50 / seconds_per_tick)))) // Roughly 50% chance per second
+			to_chat(viewer, span_userdanger("[message] [parent]!"))
+		if(prob(max(5, 100 - (80 / seconds_per_tick)))) // Roughly 20% chance per second
```

Comment:
we have delta time prob stuff, do you wanna use that instead?

Reply:
upon further thought it would be even better if this just gave you a `/datum/status_effect` rather than have the processing stuff on the component

Reply:
I thought I remembered us having helpers for this but couldn't find them in a search, keyword delta doesn't seem to help much either and most things that do something like this do it manually like here.

As for this being a status effect, it really shouldn't. Keeping track of things to remove the status effect when they leave the range will probably end up being no better performance wise, and intuitively this makes more sense as a recurring effect from an origin imo.